cmd/agent/common/misconfig/global.go

```diff
@@ -8,19 +8,25 @@ package misconfig
 import "github.com/DataDog/datadog-agent/pkg/util/log"
 
 // ToLog outputs warnings about common misconfigurations in the logs
-func ToLog() {
-	for name, check := range checks {
-		if err := check(); err != nil {
-			log.Warnf("misconfig: %s: %v", name, err)
+func ToLog(agent string) {
+	for _, check := range checks {
+		if check.Agent == agent {
+			if err := check.Run(); err != nil {
+				log.Warnf("misconfig: %s: %v", agent, err)
+			}
 		}
 	}
 }
 
 type checkFn func() error
+type check struct {
+	Agent string
```

Comment:
This should be a `map[string]struct{}`, since a single check can support multiple potential agents. For example: the `procMount` check supports both the core agent and the process agent.

The reason I say a `map[string]struct{}`, and not a `[]string` is because we can save some compute by using a set. Go unfortunately does not have a native set type so we need to use a map.

Maybe we could also rename it to `supportedAgents`?